server/app/services/program/CantAddQuestionToBlockException.java

```diff
@@ -0,0 +1,24 @@
+package services.program;
+
+import services.ProgramBlockValidation;
+import services.question.types.QuestionDefinition;
+
+public class CantAddQuestionToBlockException extends Exception {
+  public CantAddQuestionToBlockException(
+      ProgramDefinition program,
+      BlockDefinition block,
+      QuestionDefinition question,
+      ProgramBlockValidation.AddQuestionResult reason) {
+    super(
+        String.format(
+            "Can't add question to the block. Error: %s. Check comments in AddQuestionResult enum."
```

Comment:
?

Reply:
Done.
Though the error is not meant to be understandable or actionable by CiviForm admins. They are not expected to see it but if they do - they should report it to eng and we'll try to understand what went wrong. We consider adding text something like "Please report this error to ...".

As for message being atypical, I find it nice when error tries to provide more info on how to debug it. Though here I'm guessing engineers will try and find the enum value anyway.

Reply:
The comment was because we will print this in a notFound HTTP response, as I see in the PR, which the site user will see; a CiviForm Admin.  If we want someone for the IT manager then we can log that